contracts/Zauction.sol

```diff
@@ -121,6 +139,56 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     );
   }
 
+  function setBuyPrice(uint256 amount, uint256 tokenId) external {
+    address owner = registrar.ownerOf(tokenId);
+    require(msg.sender == owner, "zAuction: only owner can set price");
+    priceInfo[tokenId] = Listing(amount, owner);
```

Comment:
Add a require statement that will protect against a transaction that doesn't change state.
Or for example;
If i call this `setBuyPrice()` with the same parameters twice in a row, the second one should fail (revert)